src/image.c

```diff
@@ -31,27 +31,55 @@ typedef enum {
 #undef _
 } twin_image_format_t;
 
-/* FIXME: Check the header of the given images to determine the supported image
- * formats instead of parsing the filename without checking its content.
+/*
+ * Defines the headers of supported image formats.
+ * Each image format has a unique header, allowing the format to be
+ * determined by inspecting the file header.
+ * Supported formats: PNG, JPEG.
+ * Reference links:
+ * - PNG:
+ * http://www.libpng.org/pub/png/spec/1.2/PNG-Rationale.html#R.PNG-file-signature
+ * - JPEG:
+ * https://www.file-recovery.com/jpg-signature-format.htm
  */
+#if __BYTE_ORDER == __BIG_ENDIAN
+static const uint8_t header_png[8] = {
+    0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A,
+};
+#else
+static const uint8_t header_png[8] = {
+    0x0A, 0x1A, 0x0A, 0x0D, 0x47, 0x4E, 0x50, 0x89,
+};
+#endif
+static const uint8_t header_jpeg[3] = {0xFF, 0xD8, 0xFF};
+
 static twin_image_format_t image_type_from_name(const char *path)
```

Comment:
Rename the function `image_type_from_name` since its name is not associated with the proposed change.

Reply:
Maybe rename it as `image_type_from_header` better ?

Reply:
No, simply name it `image_type_detect`.